rbac/providers/ldap/delta_outbound_sync.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import logging
+import os
+
+import rethinkdb as r
+import ldap3
+from ldap3 import ALL, MODIFY_REPLACE, Connection, Server
+
+from rbac.providers.common.rethink_db import (
+    connect_to_db,
+    peek_at_queue,
+    put_entry_changelog,
+    delete_entry_queue,
+)
+from rbac.providers.common.outbound_filters import (
+    outbound_user_filter,
+    outbound_group_filter,
+)
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = {"OUTBOUND_QUEUE": "queue_outbound"}
+
+
+def getenv(name, default):
+    value = os.getenv(name)
+    if value is None or not value:
+        return default
+    return value
+
+
+OUTBOUND_QUEUE = getenv("OUTBOUND_QUEUE", DEFAULT_CONFIG["OUTBOUND_QUEUE"])
```

Comment:
There is quite a bit of work going on here in establishing the name of the outbound_queue. I am wondering what the intention is here. We can go over it tomorrow since I will inevitably be merging this file.